### PR TITLE
Test refactoring part 2

### PR DESCRIPTION
**Pull request by mjcarroll**
Further cleanups to testing after the subprocess changes.

This does:

* Pushes auxillary test executables down one directory, just to tidy up the `integration` folder and makes it clear which things are tests vs support executables
* Splits "test utils" (utility code) from "test config" (generated configuration)
* Puts auxillary executable definitions in one location

### CODE REVIEW

File: test/test_utils.hh [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+ * limitations under the License.
+ *
+*/
+
+#ifndef GZ_TRANSPORT_TEST_UTILS_HH_
+#define GZ_TRANSPORT_TEST_UTILS_HH_
+
+#include <climits>
+#include <filesystem>
+#include <iostream>
```
[the remark this answers is not in the document]
Comment (mjcarroll) [reply test/test_utils.hh]: removed

**Comment by mjcarroll** [comment]
@osrf-jenkins retest this please

**Comment by mjcarroll** [comment]
I found the culprit here, I accidentally switched out the test executables from one that runs indefinitely to one that exits after a few seconds.  It seems that on Linux runners, we are fast enough to complete the stress test in the window, where on other platforms we are too slow.

I have switched the test back accordingly.

**Comment by mjcarroll** [comment]
@osrf-jenkins retest this please

**Comment by caguero** [comment]
Do you know if the Windows failure is related to this PR?

**Comment by mjcarroll** [comment]
> Do you know if the Windows failure is related to this PR?

 I'm not sure.  On one hand I haven't seen it before, so maybe my fault, but also seems unrelated to the changes in the PR?

**Comment by mjcarroll** [comment]
Looks like there are actual multiple failing tests, but they aren't being reported correctly: 

```
The following tests FAILED:
	 47 - INTEGRATION_twoProcsSrvCallStress (Timeout)
	 53 - INTEGRATION_twoProcsSrvCallWithoutInputStress (Timeout)
	 59 - INTEGRATION_twoProcsSrvCallWithoutOutputStress (Timeout)
	 83 - INTEGRATION_playback (Failed)
Errors while running CTest
```

**Comment by mjcarroll** [comment]
Looks like Windows failure may have been a flake.  I will merge this and continue to monitor.

**Pull request merged**